examples/demos/create_infrastructure_with_JBOD.pp

```diff
@@ -0,0 +1,262 @@
+        enclosure_index => 3
+      }
+    ]
+  }
+}
+
+oneview_sas_logical_interconnect_group{'Test Puppet SAS LIG':
+  ensure  => 'present',
+  require => Oneview_logical_interconnect_group['Test Puppet LIG'],
+  data    => {
+    name          => 'Test Puppet SAS LIG',
+    interconnects =>
+    [
+      {
+        bay  => 1,
+        type => 'Synergy 12Gb SAS Connection Module'
+      },
+      {
+        bay  => 4,
+        type => 'Synergy 12Gb SAS Connection Module'
+      }
+    ]
+  }
+}
+
+oneview_enclosure_group{'Test Enclosure Group':
+  ensure  => 'present',
+  require => Oneview_sas_logical_interconnect_group['Test Puppet SAS LIG'],
+  data    => {
+    name                        => 'Test Enclosure Group',
+    stackingMode                => 'Enclosure',
+    interconnectBayMappingCount => '8',
+    enclosureCount              => '3',
+    interconnectBayMappings     =>
+    [
+      {
+        interconnectBay             => '1',
+        logicalInterconnectGroupUri => Oneview_sas_logical_interconnect_group['Test Puppet SAS LIG']['data']['name'],
+        enclosureIndex              => '1'
+      },
+      {
+        interconnectBay             => '4',
+        logicalInterconnectGroupUri => Oneview_sas_logical_interconnect_group['Test Puppet SAS LIG']['data']['name'],
+        enclosureIndex              => '1'
+      },
+      {
+        interconnectBay             => '1',
+        logicalInterconnectGroupUri => Oneview_sas_logical_interconnect_group['Test Puppet SAS LIG']['data']['name'],
+        enclosureIndex              => '2'
+      },
+      {
+        interconnectBay             => '4',
+        logicalInterconnectGroupUri => Oneview_sas_logical_interconnect_group['Test Puppet SAS LIG']['data']['name'],
+        enclosureIndex              => '2'
+      },
+      {
+        interconnectBay             => '1',
+        logicalInterconnectGroupUri => Oneview_sas_logical_interconnect_group['Test Puppet SAS LIG']['data']['name'],
+        enclosureIndex              => '3'
+      },
+      {
+        interconnectBay             => '4',
+        logicalInterconnectGroupUri => Oneview_sas_logical_interconnect_group['Test Puppet SAS LIG']['data']['name'],
+        enclosureIndex              => '3'
+      },
+      {
+        interconnectBay             => '3',
+        logicalInterconnectGroupUri => Oneview_logical_interconnect_group['Test Puppet LIG']['data']['name']
+      },
+      {
+        interconnectBay             => '6',
+        logicalInterconnectGroupUri => Oneview_logical_interconnect_group['Test Puppet LIG']['data']['name']
+      },
+    ]
+  }
+}
+
+oneview_logical_enclosure{'Test LE':
+  ensure  => 'present',
+  require => Oneview_enclosure_group['Test Enclosure Group'],
+  data    => {
+    name                 => 'Test LE',
+    enclosureUris        => ['/rest/enclosures/0000000000A66101', '/rest/enclosures/0000000000A66102', '/rest/enclosures/0000000000A66103'],
```

Comment:
Maybe we could use here a get all enclosures method or select by index.

Reply:
This is just a basic infrastructure and it is up to user to do the necessary extra steps. Moreover, the selection from an array may not accurately select the same set of enclosures every time.